tools/validate.js

```diff
@@ -0,0 +1,121 @@
+import { readFileSync } from 'fs';
+import { resolve } from 'path';
+
+const INTEGRATION_TYPES = ['VIRTUALIZED', 'CONTAINERIZED'];
+
+const manifestErrorMessage = (field, type) => `Value of field '${field}' should be a non-empty ${type}`;
+
+const errorMessage = ({ title, field, type, isWidget }) =>
+  `Value of field ${field} should be a non-empty ${type} in ${isWidget ? 'widget' : 'page'} "${title}"`;
+
+const isUrlValid = url => {
+  const urlRegex = /^(https?|ftp):\/\/[^\s/$.?#].[^\s]*$/i;
+  return urlRegex.test(url);
+};
+
+const validateProps = props => {
+  const errors = [`Value of field props should be an object`];
+  return props && !(typeof props === 'object') ? errors : [];
+};
+
+const checkRedundantFields = ({ config, requiredFields, optionalFields, isWidget }) =>
+  Object.entries(config).reduce((acc, [key]) => {
+    if (!requiredFields.includes(key) && !optionalFields.includes(key)) {
+      acc.push(
+        `Field '${key}' should be removed from ${
+          isWidget !== undefined ? `${isWidget ? 'widget' : 'page'} "${config.title}"` : 'manifest object'
+        }`
+      );
+    }
+    return acc;
+  }, []);
+
+const validateRequiredFields = ({ config, requiredFields, isWidget }) =>
+  requiredFields.reduce((acc, field) => {
+    if (!(field in config)) {
+      acc.push(`Field '${field}' is required in ${isWidget ? 'widget' : 'page'} "${config.title}"`);
+    } else if (field === 'scopes') {
+      if (!(Array.isArray(config[field]) && config[field].length)) {
+        acc.push(errorMessage({ title: config.title, field, type: 'array', isWidget }));
+      }
+    } else if (!(typeof config[field] === 'string' && config[field])) {
+      acc.push(errorMessage({ title: config.title, field, type: 'string', isWidget }));
+    }
+
+    return acc;
+  }, []);
+
+const validateComponent = (config, isWidget) => {
+  const requiredFields = ['id', 'title', 'url'];
+  const optionalFields = ['props'];
+
+  if (isWidget) {
+    requiredFields.push('scopes');
+  }
+
+  return [
+    ...checkRedundantFields({ config, requiredFields, optionalFields, isWidget }),
+    ...validateRequiredFields({ config, requiredFields, isWidget }),
+    ...validateProps(config.props),
+  ];
+};
+
+const validateManifestRequiredFields = (config, requiredFields) =>
+  requiredFields.reduce((acc, field) => {
+    if (!(field in config)) {
+      acc.push(`Field '${field}' is required in manifest object`);
+    } else if (field === 'integrationType' && !INTEGRATION_TYPES.includes(config[field])) {
+      acc.push(`Field ${field} should be one of ${INTEGRATION_TYPES}`);
```

Comment:
we already know if its a react type widget of iframe. user doesnt need to choose. it should be according to the widget type